src/Compiler.php

```diff
@@ -281,7 +281,7 @@ public function outdent($step = 1)
 
     public function getVarName()
     {
-        return sprintf('__internal_%s', hash('sha256', __METHOD__.$this->varNameSalt++));
+        return sprintf('__internal_%s', hash(\PHP_VERSION_ID < 80100 ? 'sha256' : 'xxh128', __METHOD__.$this->varNameSalt++));
```

Comment:
It might be possible to remove the `hash` function in all `getVarName` methods and only concat the `varNameSalt`:

```suggestion
        return sprintf('__internal_compiler_%s', $this->varNameSalt++);
```

Reply:
Done in an other PR. https://github.com/twigphp/Twig/pull/3601